ratpack-cassandra/src/main/java/smartthings/ratpack/cassandra/CassandraSessionFactory.java

```diff
@@ -0,0 +1,24 @@
+package smartthings.ratpack.cassandra;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Session;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+
+public final class CassandraSessionFactory implements Provider<Session> {
+
+	private final Cluster cluster;
+	private final CassandraModule.Config config;
+
+	@Inject
+	CassandraSessionFactory(Cluster cluster, CassandraModule.Config config) {
+		this.cluster = cluster;
+		this.config = config;
+	}
+
+	@Override public Session get() {
+		return (config.keyspace != null && !"".equals(config.keyspace)) ?
+				cluster.connect(config.keyspace) :
+				cluster.connect();
```

Comment:
Its probably my lack of understanding of Ratpack but I don't really follow what you mean.  If another component depends on a Session through dependency injection then Guice would refer to this singleton to create that.  By the nature of depending on a Session it would be available.  If `cluster.connect()` fails in this Guice provider then the app would fail to start.